dev/integration_tests/ios_add2app_life_cycle/flutterapp/lib/main.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:ui' as ui;
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_driver/driver_extension.dart';
+
+void main() {
+  print('start main');
+//  enableFlutterDriverExtension();
+  WidgetsFlutterBinding.ensureInitialized();
+  run(ui.window.defaultRouteName);
+}
+
+Future<String> run(String name) async {
+  // The platform-specific component will call [setInitialRoute] on the Flutter
+  // view (or view controller for iOS) to set [ui.window.defaultRouteName].
+  // We then dispatch based on the route names to show different Flutter
+  // widgets.
+  // Since we don't really care about Flutter-side navigation in this app, we're
+  // not using a regular routes map.
+  switch (name) {
+    case 'resumed':
+      runApp(const FlutterViewResumed());
+      break;
+    default:
+      runApp(const FlutterViewDetached());
+      break;
+  }
+  return '';
+}
+
+
+class FlutterViewResumed extends StatelessWidget {
+  const FlutterViewResumed();
+
+  @override
+  Widget build(BuildContext context) {
+    assert(SchedulerBinding.instance.lifecycleState == AppLifecycleState.resumed);
```

Comment:
@jmagman I copy the test from ios_add2app
The test is using XCTestCase to launch the app, and use tap a native button so that it will bring up a flutter view controller. The assertion is in dart side. Is there a way for dart code to notify the engine that the app has crashed?

Reply:
Sorry I just saw this.  I'm not sure now a dart assert expresses itself in an iOS process... does the app not crash, it just goes white or something?